HACKING-URL.md

```diff
@@ -0,0 +1,214 @@
+
+Introduction
+============
+
+This file contains instructions on how to deal with feature requests for
+supporting new URL domains in MusicBrainz, or to change how they are currently
+supported.
```

Comment:
* This file is rather descriptive than giving “instructions”.
* As is in this commit it rather describes the implementation of “how to deal with feature requests”; It could be useful to document the workflow for STYLE/MBS URL relationship types and lyrics/otherdatabases whitelists.
* With the exception of lyrics pages and other databases, there is generally no need of any kind of “URL domain support in MusicBrainz” for editors to add external links to new domains or even to domains that don’t exist yet. That’s why I have been using “handle” rather than “support” in commit messages for a long time.
* I suggest adding a short list of things that people may want to implement here instead of the list of how things are organized in our code you provided in the next paragraph.

```suggestion
This file describes how to implement specific handlers for external websites in MusicBrainz Server to: improve the display of links on MusicBrainz website; make it easier to add links; block unwanted links.
```

Reply:
I was thinking about blocking URL shorteners (in `externalLinks.js`) and deep-linked images (and BC profiles and whatnot in `URLCleanup.js`).